include/libcyphal/transport/udp/rx_session_tree_node.hpp

```diff
@@ -0,0 +1,121 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_UDP_RX_SESSION_TREE_NODE_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_UDP_RX_SESSION_TREE_NODE_HPP_INCLUDED
+
+#include "tx_rx_sockets.hpp"
+
+#include "libcyphal/executor.hpp"
+#include "libcyphal/transport/msg_sessions.hpp"
+#include "libcyphal/transport/session_tree.hpp"
+#include "libcyphal/transport/svc_sessions.hpp"
+#include "libcyphal/transport/types.hpp"
+#include "libcyphal/types.hpp"
+
+#include <cetl/cetl.hpp>
+#include <udpard.h>
+
+#include <array>
+#include <cstdint>
+#include <tuple>
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace udp
+{
+
+/// Internal implementation details of the UDP transport.
+/// Not supposed to be used directly by the users of the library.
+///
+namespace detail
+{
+
+template <typename SocketInterface>
+struct SocketState
+{
+    UniquePtr<SocketInterface> interface;
+    IExecutor::Callback::Any   callback;
+
+};  // SocketState
+
+class IRxSessionDelegate;
+class IMsgRxSessionDelegate;
+
+struct RxSessionTreeNode
```

Comment:
Missing class docs

Reply:
Same story here - everything about/under "delegate" is implementation details.